lib/kafka_ex/server.ex

```diff
@@ -368,21 +394,29 @@ defmodule KafkaEx.Server do
       end
 
       def update_metadata(state) do
-        {correlation_id, metadata} = retrieve_metadata(state.brokers, state.correlation_id, config_sync_timeout())
-        metadata_brokers = metadata.brokers
-        brokers = state.brokers
-          |> remove_stale_brokers(metadata_brokers)
-          |> add_new_brokers(metadata_brokers, state.ssl_options, state.use_ssl)
-        %{state | metadata: metadata, brokers: brokers, correlation_id: correlation_id + 1}
+        case retrieve_metadata(state.brokers, state.correlation_id, config_sync_timeout()) do
+          {:error, reason} ->
+            state
+
+          {correlation_id, metadata} ->
+            metadata_brokers = metadata.brokers
+            brokers = state.brokers
+              |> remove_stale_brokers(metadata_brokers)
+              |> add_new_brokers(metadata_brokers, state.ssl_options, state.use_ssl)
+            %{state | metadata: metadata, brokers: brokers, correlation_id: correlation_id + 1}
+        end
       end
 
       # credo:disable-for-next-line Credo.Check.Refactor.FunctionArity
-      def retrieve_metadata(brokers, correlation_id, sync_timeout, topic \\ []), do: retrieve_metadata(brokers, correlation_id, sync_timeout, topic, @retry_count, 0)
+      def retrieve_metadata(brokers, correlation_id, sync_timeout, topic \\ []) do
+        retrieve_metadata(brokers, correlation_id, sync_timeout, topic, @retry_count, 0)
+      end
       # credo:disable-for-next-line Credo.Check.Refactor.FunctionArity
       def retrieve_metadata(_, correlation_id, _sync_timeout, topic, 0, error_code) do
         Logger.log(:error, "Metadata request for topic #{inspect topic} failed with error_code #{inspect error_code}")
         {correlation_id, %Metadata.Response{}}
       end
+      # def retrieve_metadata({:error, message}), do: {:error, message}
```

Comment:
Commented out code.

Reply:
Apologies, I've got 1 commit that I need to rule in/out, so have grouped it all under a stash commit, 1cee8bf22f40656a27155af1a55edb074f6d9742

Reply:
I'll push into a new branch actually, no need to be here.